### PR TITLE
FOUNDATIONS: ServiceCollectionService

### LINKED ISSUE #10: FOUNDATIONS: `ServiceCollectionService`

### PR DESCRIPTION
**Pull request by LBoullosa**
closes #10

### CODE REVIEW

**Review by glhays — DISMISSED** [review]
LGTM

**Review by glhays — DISMISSED** [review]

File: STX.SPAL.Core/Services/Foundations/ServicesCollections/ServiceCollectionService.Exceptions.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                innerException: exception);
+        }
+
+        private static ServiceCollectionValidationDependencyException
+            CreateServiceCollectionValidationDependencyException(Exception exception)
+        {
+            var addServiceDescriptorException =
+               new AddServiceDescriptorException(
+                   message: "Add service descriptor error occurred, contact support.",
+                   innerException: exception);
```
Comment (cjdutoit) [inline STX.SPAL.Core/Services/Foundations/ServicesCollections/ServiceCollectionService.Exceptions.cs]: Localization should be inside the TryCatch, categorization will be in the method.  See for more info https://discord.com/channels/934130100008538142/1216159515171618887/1229894169930240130

File: STX.SPAL.Core/Services/Foundations/ServicesCollections/ServiceCollectionService.Exceptions.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+            return new ServiceCollectionValidationDependencyException(
+                    message: "Service collection validation dependency error occurred, contact support.",
+                    innerException: addServiceDescriptorException);
+        }
+
+        private static ServiceCollectionServiceException CreateServiceCollectionServiceException(
+            Exception exception)
+        {
+            var failedServiceCollectionServiceException =
```
Comment (cjdutoit) [inline STX.SPAL.Core/Services/Foundations/ServicesCollections/ServiceCollectionService.Exceptions.cs]: Localization should be inside the TryCatch, categorization will be in the method. See for more info https://discord.com/channels/934130100008538142/1216159515171618887/1229894169930240130

File: STX.SPAL.Core.Tests.Unit/Services/Foundations/ServicesCollections/ServiceCollectionServiceTests.Validations.RegisterServiceDescriptor.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                expectedServiceCollectionValidationException);
+
+            this.dependencyInjectionBroker
+                .Verify(broker =>
+                    broker.AddServiceDescriptor(It.IsAny<ServiceDescriptor>()),
+                Times.Never);
+
+            this.dependencyInjectionBroker.VerifyNoOtherCalls();
+        }
+
```
Comment (cjdutoit) [inline STX.SPAL.Core.Tests.Unit/Services/Foundations/ServicesCollections/ServiceCollectionServiceTests.Validations.RegisterServiceDescriptor.cs]: Remove empty line

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**